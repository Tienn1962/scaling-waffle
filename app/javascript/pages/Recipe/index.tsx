import {
  ArrowLeftIcon,
  ClockIcon,
  ShoppingCartIcon,
  StarIcon,
  UserIcon,
} from "@heroicons/react/outline";
import axios from "axios";
import React, { useEffect, useState } from "react";
import { Link, useParams } from "react-router-dom";

export default () => {
  const { id } = useParams();

  const [recipe, setRecipe] = useState<any>({});

  const fetchRecipe = async () => {
    const response = await axios.get(`/api/v1/recipes/${id}`);

    setRecipe(response.data.recipe);
  };

  useEffect(() => {
    fetchRecipe();
  }, []);

  return (
    <main className="flex flex-col justify-center">
      <div className="hero min-h-screen">
        <div className="hero-content rounded-lg bg-neutral">
          <div className="join join-vertical">
            <Link
              className="join-item flex btn bg-black no-animation"
              to="/recipes"
            >
              <ArrowLeftIcon className="w-6 h-6" />
              Back
            </Link>
            <img
              src={recipe.image_url}
              className="join-item max-w-sm rounded-lg shadow-2xl"
            />
          </div>

          <div>
            <h1 className="text-4xl">{recipe.title}</h1>
            {recipe.author && (
              <div className="flex">
                <UserIcon className="h-6 w-6" />
                <span className="pl-2">{recipe.author}</span>
              </div>
            )}
            <div className="flex">
              <ClockIcon className="h-6 w-6" />
              <span className="pl-2">
                {recipe.prep_time}m | {recipe.cook_time}m
              </span>
            </div>
            <div className="flex">
              <StarIcon className="h-6 w-6" />
              <span className="pl-2">{recipe.ratings}/5</span>
            </div>
            <div className="flex">
              <ShoppingCartIcon className="h-6 w-6" />
              <span className="pl-2">Ingredients :</span>
            </div>
            <ul className="m-5 flex flex-col list-disc">
              {recipe.ingredients &&
                recipe.ingredients.map((ingredient) => (
                  <li key={ingredient.title}>{ingredient.title}</li>
                ))}
            </ul>
          </div>
        </div>
      </div>
    </main>
  );
};
