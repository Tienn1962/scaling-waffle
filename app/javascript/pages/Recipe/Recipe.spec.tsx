import axios from "axios";
import React from "react";
import { render } from "@testing-library/react";
import Recipe from ".";
import { MemoryRouter } from "react-router-dom";
import { act } from "react-dom/test-utils";

jest.mock("axios");

it("renders a recipe with its ingredients", async () => {
  const recipe = {
    author: "John Recipe",
    prep_time: 10,
    cook_time: 10,
    ratings: 5.0,
    title: "Potatoes with chicken",
    ingredients: [{ title: "3 potatoes" }, { title: "chicken breast" }],
  };

  const response = { data: { recipe } };

  (axios.get as jest.Mock).mockResolvedValue(response);

  const component = await act(async () =>
    render(
      <MemoryRouter>
        <Recipe />
      </MemoryRouter>
    )
  );

  expect(component).toMatchSnapshot();
});
