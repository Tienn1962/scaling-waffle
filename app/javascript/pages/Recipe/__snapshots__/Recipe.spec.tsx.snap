// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a recipe with its ingredients 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="flex flex-col justify-center"
      >
        <div
          class="hero min-h-screen"
        >
          <div
            class="hero-content rounded-lg bg-neutral"
          >
            <div
              class="join join-vertical"
            >
              <a
                class="join-item flex btn bg-black no-animation"
                href="/recipes"
              >
                <svg
                  aria-hidden="true"
                  class="w-6 h-6"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 19l-7-7m0 0l7-7m-7 7h18"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                Back
              </a>
              <img
                class="join-item max-w-sm rounded-lg shadow-2xl"
              />
            </div>
            <div>
              <h1
                class="text-4xl"
              >
                Potatoes with chicken
              </h1>
              <div
                class="flex"
              >
                <svg
                  aria-hidden="true"
                  class="h-6 w-6"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 7a4 4 0 11-8 0 4 4 0 018 0zM12 14a7 7 0 00-7 7h14a7 7 0 00-7-7z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                <span
                  class="pl-2"
                >
                  John Recipe
                </span>
              </div>
              <div
                class="flex"
              >
                <svg
                  aria-hidden="true"
                  class="h-6 w-6"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                <span
                  class="pl-2"
                >
                  10
                  m | 
                  10
                  m
                </span>
              </div>
              <div
                class="flex"
              >
                <svg
                  aria-hidden="true"
                  class="h-6 w-6"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11.049 2.927c.3-.921 1.603-.921 1.902 0l1.519 4.674a1 1 0 00.95.69h4.915c.969 0 1.371 1.24.588 1.81l-3.976 2.888a1 1 0 00-.363 1.118l1.518 4.674c.3.922-.755 1.688-1.538 1.118l-3.976-2.888a1 1 0 00-1.176 0l-3.976 2.888c-.783.57-1.838-.197-1.538-1.118l1.518-4.674a1 1 0 00-.363-1.118l-3.976-2.888c-.784-.57-.38-1.81.588-1.81h4.914a1 1 0 00.951-.69l1.519-4.674z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                <span
                  class="pl-2"
                >
                  5
                  /5
                </span>
              </div>
              <div
                class="flex"
              >
                <svg
                  aria-hidden="true"
                  class="h-6 w-6"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3 3h2l.4 2M7 13h10l4-8H5.4M7 13L5.4 5M7 13l-2.293 2.293c-.63.63-.184 1.707.707 1.707H17m0 0a2 2 0 100 4 2 2 0 000-4zm-8 2a2 2 0 11-4 0 2 2 0 014 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                <span
                  class="pl-2"
                >
                  Ingredients :
                </span>
              </div>
              <ul
                class="m-5 flex flex-col list-disc"
              >
                <li>
                  3 potatoes
                </li>
                <li>
                  chicken breast
                </li>
              </ul>
            </div>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="flex flex-col justify-center"
    >
      <div
        class="hero min-h-screen"
      >
        <div
          class="hero-content rounded-lg bg-neutral"
        >
          <div
            class="join join-vertical"
          >
            <a
              class="join-item flex btn bg-black no-animation"
              href="/recipes"
            >
              <svg
                aria-hidden="true"
                class="w-6 h-6"
                fill="none"
                stroke="currentColor"
                stroke-width="2"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 19l-7-7m0 0l7-7m-7 7h18"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
              Back
            </a>
            <img
              class="join-item max-w-sm rounded-lg shadow-2xl"
            />
          </div>
          <div>
            <h1
              class="text-4xl"
            >
              Potatoes with chicken
            </h1>
            <div
              class="flex"
            >
              <svg
                aria-hidden="true"
                class="h-6 w-6"
                fill="none"
                stroke="currentColor"
                stroke-width="2"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16 7a4 4 0 11-8 0 4 4 0 018 0zM12 14a7 7 0 00-7 7h14a7 7 0 00-7-7z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
              <span
                class="pl-2"
              >
                John Recipe
              </span>
            </div>
            <div
              class="flex"
            >
              <svg
                aria-hidden="true"
                class="h-6 w-6"
                fill="none"
                stroke="currentColor"
                stroke-width="2"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
              <span
                class="pl-2"
              >
                10
                m | 
                10
                m
              </span>
            </div>
            <div
              class="flex"
            >
              <svg
                aria-hidden="true"
                class="h-6 w-6"
                fill="none"
                stroke="currentColor"
                stroke-width="2"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.049 2.927c.3-.921 1.603-.921 1.902 0l1.519 4.674a1 1 0 00.95.69h4.915c.969 0 1.371 1.24.588 1.81l-3.976 2.888a1 1 0 00-.363 1.118l1.518 4.674c.3.922-.755 1.688-1.538 1.118l-3.976-2.888a1 1 0 00-1.176 0l-3.976 2.888c-.783.57-1.838-.197-1.538-1.118l1.518-4.674a1 1 0 00-.363-1.118l-3.976-2.888c-.784-.57-.38-1.81.588-1.81h4.914a1 1 0 00.951-.69l1.519-4.674z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
              <span
                class="pl-2"
              >
                5
                /5
              </span>
            </div>
            <div
              class="flex"
            >
              <svg
                aria-hidden="true"
                class="h-6 w-6"
                fill="none"
                stroke="currentColor"
                stroke-width="2"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 3h2l.4 2M7 13h10l4-8H5.4M7 13L5.4 5M7 13l-2.293 2.293c-.63.63-.184 1.707.707 1.707H17m0 0a2 2 0 100 4 2 2 0 000-4zm-8 2a2 2 0 11-4 0 2 2 0 014 0z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
              <span
                class="pl-2"
              >
                Ingredients :
              </span>
            </div>
            <ul
              class="m-5 flex flex-col list-disc"
            >
              <li>
                3 potatoes
              </li>
              <li>
                chicken breast
              </li>
            </ul>
          </div>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
